lib/tasks/fix_broken_links_with_url_redirection.rake

```diff
@@ -0,0 +1,69 @@
+namespace :OH do
+  desc 'Fix broken links failed with "301: Net::HTTPMovedPermanently" error'
+
+  task fix_broken_links_with_url_redirection: :environment do
+    @max_limit = 10
+    @valid_response_codes = %w[200 301 308]
+    @log = Logger.new('log/fixed_link_with_url_redirection.log')
+
+    BrokenLink.where(error: ['301: Net::HTTPMovedPermanently', '308: Net::HTTPRedirection']).each do |broken_link|
+      @link = broken_link.link
+      @log.info("#{@link.id}, #{@link.url}")
+      code, new_url = fetch_response(@link.url)
+      broken_link.destroy && next if code == '200'
+      handle_url_redirection(new_url)
+    end
+  end
+
+  def fetch_response(url)
+    uri = URI(url)
+    request = Net::HTTP::Head.new(uri.request_uri)
+    response = http_object(uri).request(request)
+    return unless response && @valid_response_codes.include?(response.code)
+    [response.code, response['location']]
+  rescue
+    false
+  end
+
+  def http_object(uri)
+    http = Net::HTTP.new(uri.host, uri.port)
+    http.read_timeout = 30
+    http.open_timeout = 30
+    http.use_ssl = (uri.scheme == 'https')
+    http.verify_mode = OpenSSL::SSL::VERIFY_NONE if uri.scheme == 'https'
+    http
+  end
+
+  def handle_url_redirection(url, limit = 1)
+    return if url.blank? || @link.url == url || limit > @max_limit
```

Comment:
Given that there is a max limit defined during initialization, this line could his that max limit and the `handle_url_redirection` could fail silently, but this seems like a small risk that doesn't warrant additional protections.